cmd/registry/patch/spec.go

```diff
@@ -84,26 +86,55 @@ func applyApiSpecPatch(
 		},
 		AllowMissing: true,
 	}
-	// TODO: add support for multi-file specs
-	// TODO: add support for local file import (maybe?)
 	// TODO: verify mime type
 	if spec.Data.SourceURI != "" {
-		resp, err := http.Get(spec.Data.SourceURI)
+		u, err := url.ParseRequestURI(spec.Data.SourceURI)
 		if err != nil {
 			return err
 		}
-		body, err := ioutil.ReadAll(resp.Body)
-		if err != nil {
-			return err
-		}
-		if strings.Contains(spec.Data.MimeType, "+gzip") {
-			body, err = core.GZippedBytes(body)
+		if u.Scheme == "http" || u.Scheme == "https" {
```

Comment:
Consider using a switch statement for this.

Reply:
SGTM, changed.